File: app/src/main/java/paul/barthuel/go4lunch/data/local/UserSearchRepository.java
package paul.barthuel.go4lunch.data.local;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class UserSearchRepository {

    private static UserSearchRepository sInstance;

    public static UserSearchRepository getInstance() {
        if(sInstance == null) {
            sInstance = new UserSearchRepository();
        }
        return sInstance;
    }

    private final MutableLiveData<String> userSearchQueryLiveData = new MutableLiveData<>();

    private UserSearchRepository() {

    }

    public void updateSearchQuery(String newText) {
        userSearchQueryLiveData.setValue(newText);
    }

    public LiveData<String> getUserSearchQueryLiveData() {
        return userSearchQueryLiveData;
    }
}

File: app/src/main/java/paul/barthuel/go4lunch/data/local/UserSearchRepository.kt
package paul.barthuel.go4lunch.data.local

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData

class UserSearchRepository private constructor() {
    val userSearchQueryLiveData = MutableLiveData<String>()
    fun updateSearchQuery(newText: String) {
        userSearchQueryLiveData.value = newText
    }

    fun getUserSearchQueryLiveData(): LiveData<String> {
        return userSearchQueryLiveData
    }

    companion object {
        private var sInstance: UserSearchRepository? = null
        val instance: UserSearchRepository
            get() {
                if (sInstance == null) {
                    sInstance = UserSearchRepository()
                }
                return sInstance!!
            }
    }
}